src/nvim/runtime.c

```diff
@@ -2014,7 +2014,7 @@ void cmd_source_buffer(const exarg_T *const eap, bool ex_lua)
   };
   if (ex_lua || strequal(curbuf->b_p_ft, "lua")
       || (curbuf->b_fname && path_with_extension(curbuf->b_fname, "lua"))) {
-    char *name = ex_lua ? ":lua (no file)" : ":source (no file)";
+    char *name = ex_lua ? ":{range}lua" : ":source (no file)";
```

Comment:
`:lua (no file)` is misleading because `:lua` never takes a file arg.

I also considered changing it to just `:lua`, because `:{range}lua` is a bit noisy to read in a message.